include/deal.II/base/mpi.h

```diff
@@ -498,6 +498,39 @@ namespace Utilities
                                 const MPI_Comm &comm);
 #endif
 
+
+    /**
+     * Create an MPI_Datatype that consists of @p n_bytes bytes.
+     *
+     * The resulting data type can be used in MPI send/recv or MPI IO to process
+     * messages of sizes larger than 2 GB with MPI_Byte as the underlying data
+     * type. This helper is required for MPI versions before 4.0 because
+     * routines like MPI_Send
+     * use a signed interger for the @p count variable. Instead, you can use this
+     * data type with the appropriate size set to the size of your message and
+     * by passing
+     * 1 as the @p count.
+     *
+     * @note You need to free this data type after you are done using it by calling
+     * <code>MPI_Type_free(&result)</code>.
+     *
+     * Usage example:
+     * <code>
+     * std::vector<char> buffer;
+     * if (buffer.size()<(1U<<31))
+     *   MPI_Send(buffer.data(), buffer.size(), MPI_BYTE, dest, tag, comm);
+     * else
+     * {
+     *   MPI_Datatype bigtype;
+     *   Utilities::MPI::create_mpi_data_type_n_bytes(bigtype, buffer.size());
+     *   MPI_Send(buffer.data(), 1, bigtype, dest, tag, comm);
+     *   MPI_Type_free(&bigtype);
+     * }
+     * </code>
+     */
+    void
+    create_mpi_data_type_n_bytes(MPI_Datatype &result, std::size_t n_bytes);
```

Comment:
My preference would be if you returned the data type, rather than via an argument. Alternatively, if you want to keep it as you have it here, please make the `n_bytes` argument `const` and move it to first place.